### PR TITLE
Added interface for Iterator

### PR DESCRIPTION
**Pull request by yurykozyrev**

### CODE REVIEW

File: pqtgo/generator.go [hunk 1]
```diff
@@ -310,6 +310,14 @@ type %sIterator struct {
...
 	cols []string
 }
+type I%sIterator interface {
+	Next() bool
+	Close() error
+	Err() error
+	Columns() ([]string, error)
+	%s() (*%sEntity, error)
+}
+var _ I%sIterator = (*%sIterator)(nil)
```
Comment (piotrkowalczuk) [inline pqtgo/generator.go]: Whats the purpose of it?

File: pqtgo/generator.go [hunk 2]
```diff
@@ -310,6 +310,14 @@ type %sIterator struct {
 	rows *sql.Rows
 	cols []string
 }
+type I%sIterator interface {
```
Comment (piotrkowalczuk) [inline pqtgo/generator.go]: Maybe it's a good idea to generate for each entity `<entity-name>Iterator` interface and struct that will implement this interface rename to `<entity-name>Rows`? We could avoid having naming workarounds and new struct name will correspond to [sql.Rows](https://golang.org/pkg/database/sql/#Rows)

File: pqtgo/generator.go (hunk 1 above)
Comment (yurykozyrev) [inline pqtgo/generator.go]: check on compile time that struct fits the interface, I have found this approach in Google libraries, like this https://github.com/grpc/grpc-go/blob/master/server.go#L549

File: pqtgo/generator.go [hunk 3]
```diff
@@ -305,26 +305,34 @@ func (g *Generator) generateIterator(w io.Writer, t *pqt.Table) {
 	entityName := g.Formatter.Identifier(t.Name)
 	fmt.Fprintf(w, `
 
-// %sIterator is not thread safe.
-type %sIterator struct {
+// %sTypedRows are not thread safe.
```
Comment (piotrkowalczuk) [inline pqtgo/generator.go]: I think it could be simply `%sRows`.

File: pqtgo/generator.go (hunk 1 above)
Comment (piotrkowalczuk) [inline pqtgo/generator.go]: Cool, I like it.

**Comment by piotrkowalczuk** [comment]
Example app needs to be regenerated. 

```
cd example
make gen
make test
```

**Comment by yurykozyrev** [comment]
See #11 for updates

**Pull request closed without merging**